.github/workflows/maven.yml

```diff
@@ -0,0 +1,41 @@
+name: Java CI with Maven
+
+on: [push, pull_request]
```

Comment:
Do we want to do this on any/all pushes and PRs?

We could limit this to just `sprint*-staging` and master/main via `$default-branch`.

Reply:
I prefer all pull requests and pushes. That is parity with previous CI.